src/NDItk/multigroup/Structure/src/read.hpp

```diff
@@ -0,0 +1,16 @@
+/**
+ *  @brief Read the record data
+ *
+ *  This function overrides the base record read() function to enable
+ *  verification of the read data.
+ *
+ *  @param[in,out] iter     an iterator to the current position in the input
+ *  @param[in,out] end      an iterator to the end of the input
+ *  @param[in,out] groups   the number of groups to be read
```

Comment:
Number of groups + 1

Reply:
Yes. I fixed this in fix/review some time ago.